src/coreclr/jit/lclvars.cpp

```diff
@@ -4762,7 +4760,7 @@ void Compiler::lvaComputeRefCounts(bool isRecompute, bool setSlotNumbers)
             varDsc->lvSingleDef             = varDsc->lvIsParam;
             varDsc->lvSingleDefRegCandidate = varDsc->lvIsParam;
 
-            varDsc->lvAllDefsAreNoGc = true;
+            varDsc->lvAllDefsAreNoGc = (varDsc->lvImplicitlyReferenced == false);
```

Comment:
This could also be just a `lvIsOSRLocal` but went with more boarded one. Also it seems to better describe the reasoning.